### PR TITLE
Extract Disjoint Pool into static lib

### PR DESCRIPTION
**Pull request by PatKamin**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -16,6 +16,7 @@ include(helpers)
 
 # Build Options
 option(UMF_BUILD_SHARED_LIBRARY "Build UMF as shared library" OFF)
+option(UMF_BUILD_LIBUMF_POOL_DISJOINT "Build the libumf_pool_disjoint static library" OFF)
```
Comment (bratpiorka) [inline CMakeLists.txt:19]: please update the README file

**Review by bratpiorka — CHANGES_REQUESTED** [review]

File: src/pool/CMakeLists.txt
Comment (vinser52) [inline src/pool/CMakeLists.txt:1]: As I understand, with this PR C++ is only required to build DisjointPool. The core UMF library is in pure C, right?
If so, could you please update the CMake file accordingly and change the build requirements in the Readme file?

**Review by igchor — APPROVED** [review]

File: src/pool/CMakeLists.txt
Comment (PatKamin) [reply src/pool/CMakeLists.txt:1]: Not yet. There is a file `provider_tracking_windows.cpp` which is not a part of Disjoint Pool nor tests. For now we still have to require C++ for UMF build as this is one of the core UMF files (on Windows). Shall this file be rewritten in C in the future?

Comment (bratpiorka) [reply src/pool/CMakeLists.txt:1]: @PatKamin yes, we should rewrite it if needed. I will add a task for this

File: src/pool/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Copyright (C) 2023 Intel Corporation
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# libumf_pool_disjoint
+if(UMF_BUILD_LIBUMF_POOL_DISJOINT)
+    add_umf_library(disjoint_pool STATIC
+        pool_disjoint.cpp)
+
+    add_library(${PROJECT_NAME}::disjoint_pool ALIAS disjoint_pool)
```
Comment (bratpiorka) [inline src/pool/CMakeLists.txt:10]: why don't you use add_umf_library? Or better - update this function and use it also for pool_jemalloc

**Review by bratpiorka — CHANGES_REQUESTED** [review]

File: src/pool/CMakeLists.txt (hunk 2 above)
Comment (PatKamin) [reply src/pool/CMakeLists.txt:10]: I don't use this custom wrapper of `add_library` as on aliased targets `target_include_directories` cannot be called. Beside that, note that `add_library(name ALIAS target)` is not the same function as ie. `add_library(name STATIC files)`. The former just creates an alias (another name) for an already existing CMake target whereas the latter creates a brand new CMake target.

I don't want to include aliasing into the `add_umf_library` function as well as we don't need all of our targets to have nice aliases with a namespace prefix, only those that are meant to be public and exposed to users when including the UMF repo with ie. FetchContent(). These I think should be limited to targets including headers and libraries containing the UMF API.

I believe that `pool_jemalloc` should be aliased, too. However, I save it for the more general UMF installation refactoring in another PR.

**Comment by PatKamin** [comment]
Rebased

File: src/pool/CMakeLists.txt (hunk 2 above)
Comment (bratpiorka) [reply src/pool/CMakeLists.txt:10]: ok thanks for the clarification. Regarding  aliasing pool_jemalloc - I will add a task about that

**Review by bratpiorka — APPROVED** [review]

**Review by KFilipek — APPROVED** [review]
LGTM

File: src/pool/CMakeLists.txt
Comment (bratpiorka) [reply src/pool/CMakeLists.txt:1]: @vinser52 if you think everything is ok could you please close this conversation?

File: .github/workflows/basic.yml [hunk 3]
```diff
@@ -45,6 +45,7 @@ jobs:
           -DUMF_FORMAT_CODE_STYLE=OFF
           -DUMF_DEVELOPER_MODE=ON
           -DUMF_BUILD_LIBUMF_POOL_JEMALLOC=ON
+          -DUMF_BUILD_LIBUMF_POOL_DISJOINT=ON
           ${{matrix.pool_tracking}}
 
       - name: Build UMF
```
Comment (ldorau) [inline .github/workflows/basic.yml:51]: Could you add (as the last one) the new step for testing `make install`?
`- name: Install UMF`

**Review by ldorau — CHANGES_REQUESTED** [review]
Will you add the `uninstall` target later?

**Comment by ldorau** [comment]
@bratpiorka  BTW: later there should be added a *.pc (pkgconfig) file for each library.

File: .github/workflows/basic.yml (hunk 3 above)
Comment (bratpiorka) [reply .github/workflows/basic.yml:51]: I think this should be a part of separate PR where we add some tests for the installation. There is a tasks for that in our JIRA [UMFW-176](https://jira.devtools.intel.com/browse/UMFW-176)

**Comment by PatKamin** [comment]
> Will you add the `uninstall` target later?

This might be added in another PR.

File: .github/workflows/basic.yml (hunk 3 above)
Comment (ldorau) [reply .github/workflows/basic.yml:51]: OK

**Review by ldorau — APPROVED** [review]

**Comment by ldorau** [comment]
@PatKamin Rebase please

**Comment by PatKamin** [comment]
> @PatKamin Rebase please

Rebased

**Comment by PatKamin** [comment]
> @bratpiorka BTW: later there should be added (and installed) a *.pc (pkgconfig) file for each library.

I'm not sure if we need them if we have CMake config file? I guess that serves the same purpose?

**Pull request merged**